index.js

```diff
@@ -20,35 +22,293 @@ class J5sExecutor extends Executor {
      */
     _jenkinsCommand(options, callback) {
         // To pass arguments as an array, we need to use apply
-        options.params.push(callback);
-
-        return this.jenkinsClient[options.module][options.action].apply(
+        this.jenkinsClient[options.module][options.action].apply(
             this.jenkinsClient[options.module],
-            options.params);
+            options.params.concat([callback]));
+    }
+
+    /**
+     * Create or update Jenkins job
+     * @method _jenkinsJobCreateOrUpdate
+     * @param  {String}   jobName            Jenkins job name
+     * @param  {String}   xml                Jenkins job configuration
+     * @return {Promise}
+     */
+    _jenkinsJobCreateOrUpdate(jobName, xml) {
```

Comment:
I don't really understand why every method in this class starts with `_`. There's nothing about this that makes the method private, it just makes things difficult to type and read. Also, if these are meant to be private, you should add `@private` to the methods jsdoc.